Project2/Part_B/Deck.cpp

```diff
@@ -59,7 +59,28 @@ void Deck::shuffle(void)
   }
 }
 
-std::ostream& operator<<(std::ostream& os, Deck rhs)
+Card* Deck::deal(void)
+{
+  Card* dealt = this->header;
+  this->header = dealt->getNext();
+  dealt->setNext(nullptr);
+  return dealt;
+}
+
+void Deck::replace(Card* card)
+// this is a bit ugly but worked best for now while the copy constructor and assignment operator
+// are both in a different branch
+{
+  Card* head = this->header;
+  while (head->getNext() != nullptr) {
+    head = head->getNext();
+  }
+  head->setNext(card);
+  card->setNext(nullptr);
+  return;
```

Comment:
this is a function of type `void`, having `return;` at the end does exactly zero, zip, zilch, nada, nothing